### PR TITLE
Add HTMLImageElement#sizes. Also tighten up the authoring requirements f...

### PR DESCRIPTION
**Pull request by zcorpan**
...or the various attributes.

### CODE REVIEW

**Comment by nwtn** [comment]
Wait, why no `type` attr? The "Updating the List of Source Sets" algo allows for one.

**Comment by zcorpan** [comment]
Oh, I didn't realize the algorithm looked at `type`. Hmm...

File: index.src.html [hunk 1]
```diff
@@ -940,14 +940,41 @@ <h3 id='picture-idl'>
...
+	element as its parent, may also have the
+	<dfn element-attr for="source">media</dfn> attributes specified. If it
+	is specified, the value must contain a <a>valid media query</a>.
+
+	The <a element>source</a> element, if it has a <a element>picture</a>
+	element as its parent, must not have the
+	<a element-attr for="source">src</a> attribute specified.
+
+	The <dfn element-attr for="source">type</dfn> attribute, when specified
+	on a <a element>source</a> element that it has a <a element>picture</a>
```
Comment (yoavweiss) [inline index.src.html]: s/it// ?

File: index.src.html [hunk 2]
```diff
@@ -940,14 +940,41 @@ <h3 id='picture-idl'>
...
+	element as its parent, gives the type of the images in the <a>source
+	set</a>, to allow the user agent to skip to the next
+	<a element>source</a> if it does not support the given type.
+
+	<p class='note'>If the <a element-attr for='source'>type</a> attribute
+	is <em>not</em> specified, the user agent will not select a different
+	<a element>source</a> element if it finds that it does not support
+	the image format after fetching it.
```
Comment (yoavweiss) [inline index.src.html:988]: Good point

**Comment by yoavweiss** [comment]
Looks good. Can you rebase, so it can be merged?

**Pull request merged**